app/routes.js

```diff
@@ -65,8 +74,10 @@ export const requireAuthAndNoPatient = (api) => (nextState, replace, cb) => {
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireNoAuth = (api) => (nextState, replace) => {
-  if (api.user.isAuthenticated()) {
+export const requireNoAuth = (api, store) => (nextState, replace) => {
+  let { blip: state } = store.getState();
+
+  if (state.isLoggedIn || api.user.isAuthenticated()) {
```

Comment:
Same comment as I made in `requireAuth`. Also it feels dangerous here to use `state.isLoggedIn || api.user.isAuthenticated()`...surely we want the `api.user.isAuthenticated()` to be the source of truth, and not allow some potential weird state mismatch to make it look as if you're logged in?